src/components/utils/index.js

```diff
@@ -35,12 +35,20 @@ const enumerateDevicesInternal = (onSuccess, onError) => {
   }
 }
 
-export const checkIfHasWebcam = onResult => {
-  enumerateDevicesInternal(
-    devices => onResult( devices.some(device => device.kind === "videoinput") ),
-    () => onResult(false)
-  )
-}
+const checkDevicesInfo = checkFn =>
+  onResult =>
+    enumerateDevicesInternal(
+      devices => onResult(checkFn(devices)),
+      () => onResult(false)
+    )
+
+export const checkIfHasWebcam = checkDevicesInfo(
+  devices => devices.some(({ kind }) => kind === 'videoinput')
```

Comment:
I think some browsers give out an inconsistent `kind`, some give `videoinput`, some give `video`. Might be worth just checking if it contains `"video"`

Reply:
Hmm I see. Will update. Thanks!